### PR TITLE
feat(2087): Clear Cache implementation

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context
Cache cleanup buttons in Pipeline settings page, does not clear the cache when using disk cache.

## Objective
This PR implements a cache clear api for s3 or disk strategy and routes accordingly to store or queue service for further processing.

## References
https://github.com/screwdriver-cd/screwdriver/issues/2087

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/caches/request.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * @param {String} request.method
+ * @param {String} request.path
+ * @param {Object} request.payload
+ * @param {Object} request.headers
+ */
+async function invoke(request) {
+    const { method, path, payload, auth, params } = request;
+    const { store, queue, cache } = ecosystem;
+
+    const apiPath = path.replace(/\/v4/, '/v1');
```
Comment (DekusDenial) [inline plugins/caches/request.js:30]: would be nice if we can get it from API config somewhere

File: plugins/caches/request.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    const options = {
+        json: true,
+        method,
+        uri: `${store}${apiPath}`,
+        headers: {
+            Authorization: `Bearer ${auth.token}`,
+            'Content-Type': 'application/json'
```
Comment (DekusDenial) [inline plugins/caches/request.js:37]: u dont need this if `json: true` right?

File: plugins/caches/delete.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            strategies: ['token'],
+            scope: ['user', 'pipeline', '!guest']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: async (request, reply) => {
+            const res = await api.invoke(request);
```
Comment (DekusDenial) [inline plugins/caches/delete.js:22]: if the request fails, even after retries, will this handler handle that error and pass error message back to the UI to display?

Comment (pritamstyz4ever) [reply plugins/caches/delete.js:22]: Pass, the message back to the UI. Also, I think the UI should show whether cache was cleared or not. Currently it doesn't do anything

File: plugins/stats.js [hunk 4]
```diff
@@ -24,11 +24,15 @@ exports.register = (server, options, next) => {
                 schema: schema.api.stats
             }
         },
-        handler: (request, reply) =>
+        handler: async (request, reply) => {
+            const executorStatsresponse = await executor.stats({ token: '' });
+            const executorStats = executorStatsresponse.body;
```
Comment (DekusDenial) [inline plugins/stats.js:29]: what is this change for?

File: plugins/caches/delete.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+            return reply(res).code(res.statusCode);
+        }
+    }
+});
```
Comment (tkyi) [inline plugins/caches/delete.js:27]: can you add param validation?

File: plugins/stats.js (hunk 4 above)
Comment (pritamstyz4ever) [reply plugins/stats.js:29]: stats api was broken after the queue service change, this change should give the correct stats result of executor and scm

Comment (DekusDenial) [reply plugins/stats.js:29]: oh so this is separate

Comment (jithine) [reply plugins/stats.js:29]: does this works in a setup where `queue-service` is not used ?

File: plugins/pipelines/caches/request.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+ * @param {Object} request.headers
+ */
+async function invoke(request) {
+    const { method, payload, query, auth, params } = request;
+    const pipelineId = params.id;
+    const { store, queue, cache } = request.server.app.ecosystem;
+    const { scope, cacheId } = query;
+    const { username, scmContext } = auth.credentials;
+
+    console.log(username);
```
Comment (jithine) [inline plugins/pipelines/caches/request.js:31]: remove console.logs

**Pull request merged**